Goal: exit without error from smbstatus when no connections have
     been seen yet

Fixes: #164179. Patch later corrected in #164489

Status wrt upstream: Should be forwarded

Note: 

Index: samba-3.0.28a/source/utils/status.c
===================================================================
--- samba-3.0.28a.orig/source/utils/status.c
+++ samba-3.0.28a/source/utils/status.c
@@ -377,6 +377,16 @@
 			tdb_close(tdb);
 		}
 
+		tdb = tdb_open_log(lock_path("locking.tdb"), 0, TDB_DEFAULT, O_RDONLY, 0);
+
+		if (!tdb) {
+			d_printf("%s not initialised\n", lock_path("locking.tdb"));
+			d_printf("This is normal if an SMB client has never connected to your server.\n");
+			exit(0);
+		} else {
+			tdb_close(tdb);
+		}
+
 		if (!locking_init(1)) {
 			d_printf("Can't initialise locking module - exiting\n");
 			exit(1);
